### PR TITLE
[RLlib] Remove deprecated execution plan API code.

### PR DESCRIPTION
**Pull request by sven1977**
Signed-off-by: sven1977 <svenmika1977@gmail.com>

* Remove deprecated execution plan API code as far as possible.
* MB-MPO and MAML remain the only algos using the execution plan API.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm.py [hunk 1]
```diff
@@ -164,9 +164,9 @@ class Algorithm(Trainable):
 
     You can write your own Algorithm classes by sub-classing from `Algorithm`
     or any of its built-in sub-classes.
-    This allows you to override the `execution_plan` method to implement
+    This allows you to override the `training_step` method to implement
     your own algorithm logic. You can find the different built-in
-    algorithms' execution plans in their respective main py files,
+    algorithms' `training_step`s in their respective main .py files,
```
Comment (avnishn) [inline rllib/algorithms/algorithm.py:169]: Replace training_steps w `training_step` functions

**Review by avnishn — APPROVED** [review]
Awesome possum. Just one tiny bit about a grammar thing. Feel free to ignore I think people will still understand.

File: rllib/algorithms/algorithm.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/algorithm.py:169]: Good catch, fixed.

**Comment by cadedaniel** [comment]
@sven1977 This broke LinkCheck
Screen Shot 2022-11-02 at 11 04 47 AM

**Pull request merged**